src/Core/Notifications/INotificationService.cs

```diff
@@ -18,10 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using System.Linq;
+using SonarLint.VisualStudio.Core.InfoBar;
+
 namespace SonarLint.VisualStudio.Core.Notifications
 {
-    public interface INotificationService
+    public interface INotificationService : IDisposable
     {
         void ShowNotification(INotification notification);
     }
+    
+    public sealed class NotificationService : INotificationService
```

Comment:
Implementation similar to https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/IssueViz.Security/OpenInIDE/Api/OpenInIDEFailureInfoBar.cs